packages/babel-generator/src/node/parentheses.ts

```diff
@@ -473,13 +473,11 @@ export function Identifier(
   // ECMAScript specifically forbids a for-of loop from starting with the
   // token sequence `for (async of`, because it would be ambiguous with
   // `for (async of => {};;)`, so we need to add extra parentheses.
-  //
-  // If the parent is a for-await-of loop (i.e. parent.await === true), the
-  // parentheses aren't strictly needed, but we add them anyway because
-  // some tools (including earlier Babel versions) can't parse
-  // `for await (async of [])` without them.
   return (
-    node.name === "async" && isForOfStatement(parent) && node === parent.left
+    node.name === "async" &&
+    isForOfStatement(parent) &&
+    node === parent.left &&
+    !parent.await
```

Comment:
```suggestion
    isForOfStatement(parent, { left: node, await: false })
```
If for-await was introduced before Babel 7 so that every parser ^7.0.0 consistently set `await` to `false`.

Reply:
We should probably at compile time replace the object param to the various `is*` function with individual checks, and check if it makes any perf difference when transforming some medium-sized AST